pkg/releaselock/controller.go

```diff
@@ -94,114 +117,232 @@ func NewLockReleaseController(client kubernetes.Interface, config *LockReleaseCo
 	return lc, nil
 }
 
-func (c *LockReleaseController) Run(ctx context.Context) {
-	run := func(ctx context.Context) {
-		klog.Infof("Lock release controller %s started leading on node %s", c.id, c.hostname)
-		wait.Forever(func() {
-			start := time.Now()
-			cmList, err := c.client.CoreV1().ConfigMaps(util.ManagedFilestoreCSINamespace).List(ctx, metav1.ListOptions{})
-			duration := time.Since(start)
-			c.RecordKubeAPIMetrics(err, metrics.ConfigMapResourceType, metrics.ListOpType, metrics.ReconcilerOpSource, duration)
-			if err != nil {
-				klog.Errorf("Failed to list configmap in namespace %s: %v", util.ManagedFilestoreCSINamespace, err)
-				return
-			}
-			klog.Infof("Listed %d configmaps in namespace %s", len(cmList.Items), util.ManagedFilestoreCSINamespace)
-
-			start = time.Now()
-			nodes, err := c.listNodes(ctx)
-			duration = time.Since(start)
-			c.RecordKubeAPIMetrics(err, metrics.NodeResourceType, metrics.ListOpType, metrics.ReconcilerOpSource, duration)
-			if err != nil {
-				klog.Errorf("Failed to list nodes: %v", err)
-				return
-			}
-			klog.Infof("Listed %d nodes", len(nodes))
-
-			for _, cm := range cmList.Items {
-				// Filter out root ca.
-				if cm.Name == rootCA {
-					continue
-				}
-				if err := c.syncLockInfo(ctx, &cm, nodes); err != nil {
-					klog.Errorf("Failed to sync lock info for configmap %s/%s: %v", cm.Namespace, cm.Name, err)
-				}
-			}
-		}, c.config.SyncPeriod)
-	}
-
-	rl, err := resourcelock.New(
-		resourcelock.LeasesResourceLock,
-		util.ManagedFilestoreCSINamespace,
-		leaseName,
-		nil,
-		c.client.CoordinationV1(),
-		resourcelock.ResourceLockConfig{
-			Identity: c.id,
-		})
-	if err != nil {
-		klog.Fatalf("Error creating resourcelock: %v", err)
+func (c *LockReleaseController) Run(ctx context.Context) error {
+	defer utilruntime.HandleCrash()
+	defer c.updateEventQueue.ShutDown()
+	defer c.createEventQueue.ShutDown()
+	klog.Infof("Lock release controller %s started leading on node %s", c.GetId(), c.GetHost())
+	if !cache.WaitForCacheSync(ctx.Done(), (*c.nodeInformer).HasSynced) {
+		klog.Fatal("Timed out waiting for caches to sync")
+		return fmt.Errorf("failed to wait for caches to sync")
```

Comment:
I don't think this line will ever be executed.

Reply:
Removed it